release/src/router/email-3.1.3/email.1
.TH EMAIL 1

.SH NAME
email \- Encrypted SMTP email via Command line

.SH SYNOPSIS

email [options] recipient1,recipient2,recipient3,...

.SH DESCRIPTION

Email is a command line email client similiar to 'mailx'.
Added features make this a more advanced client for
sending email via the command line.  Email works with
sendmail, just as 'mailx'
does, but email also allows sending to remote smtp
servers for your email delivery.  It also works with
GPG for encrypting and signing the emails on the fly.

Email also incorporates a few other features as well.
These features include signature files with wildcard
options to specify the data, time, day, version,
system, and even a 'fortune' with the help of the
.I 'fortune' 
command.  Email also supports a very
configurable address book.  This way you can specify
a persons name with an email address and also place
a group of names into one group for sending.

Email also supports attachments.  It will accept N attachments
via a command line option and encode them with Base64 before
sending them with the email.  

Email works with a configuration file named email.conf
which is found in /etc/email/email.conf
although can be changed by specifying the --sysconfdir option
during the ./configure.  If you do not have or want a 
configuration file, you can specify the -r option to 
specify your smtp server and this will allow you to bypass
using a configuration file and use default values. If you
specify the -r option (listed below) and you do have a 
configuration file, it will still use the configuration
file but override the SMTP_SERVER variable with what is
used at the command line.  

.SH OPTIONS
There are a few possible options to email.  I have listed
them in order of relevance of usage:
.TP
.B \-\-help module | \-h "Module of Help"
This option can be specified with a module of help
topic, or without one.  Without a module of help topic,
it will just print a standard help screen with all
your possible options.  With a module of help option,
it will display a more detailed discription of that
module.

Modules are determined by command line switches.
For instance:  if you want to know about the 'encrypt'
command line option, you would specify \-\-help 'encrypt'
or \-\-help 'e' and help will display the correct module
help section.

The Help is not statically programmed into email.
Instead it is a file in email's home directory called
'email.help'.  It is updated regularly and will always
be rewritten with every release of email.

.TP
.B \-\-verbose | \-V
This option will display the progress of communicating
with the SMTP server. A pretty progress bar will be 
displayed when data is sent. This option is replacing
the old \-quiet option that would do the exact opposite.

.TP
.B \-\-blank-mail | \-blank-mail | \-b
Use this option if you would like to send a blank email
from the command line.  This is good if you just want to 
send a message with only the headers and no content.

.TP
.B \-\-subject subject | \-subject subject | \-s "Subject of Email"
This option should be rather obvious.  You specify the
subject of the email with this option.  If you are not
redirecting to standard input and you do not specify a
subject, email will ask you for the subject.  Null
subjects are allowed.

.TP
.B \-\-encrypt | \-encrypt | \-e  
This options allows you to encrypt your email message
using 'gpg'. GPG can be obtained by going to:
.URL http://www.gnupg.org.
When this option is specified, it will get the first
recipient from the recipient list and it will encrypt
the message using their key.  This key MUST be present
in order for the encryption to work properly.

.TP
.B \-\-sign | \-sign
This options will let you sign your emails
on the fly... It uses the 'gpg --detach-sign' option.
It will 'sign' the email with your public key.

.TP
.B \-\-cc recipient | \-cc recipient1,recipient2,recipient3,...
This option allows you to "curtosy copy" a list of
recipients. Recipients can be from the address book
or just plain email addresses.  They should be comma
delimited just as the main recipients will be.  CC
recipients *will* be posted in the headers and read
by email clients.

.TP
.B \-\-bcc receipient | \-bcc recipient1,recipient2,recipient3,...
Same as the --cc option, but these recipients will
.B *not* 
be posted in the headers of the email.  This
is a positive solution to "secretly"
copy someone on the email without the other recipients
knowing so (Managers usually bcc their boss when
sending an email to you about your performance and
you'll never know it 
.B)
)

.TP
.B \-\-attach file | \-a file
Attach a binary file.  This option will allow you 
to attach N files to your email and will be base64
encoded upon delivery.  This option is helpful if
you need to send any type of file that is not plain
ascii text. This option can be used multiple times
to add multiple files to attach.

.TP
.B \-\-smtp-server server | \-r server
This option will let you override the smtp server
that is specified in the email.conf file... You can
use this option if you do not have an email.conf
file.  It will allow you to use email without a 
configuration file. A helpful option to add with
this option would be the \-\-smtp-port option, 
however, it is not needed.

.TP
.B \-\-smtp-port port | \-p port
This option works well with the above option as it
will allow you to bypass the configuration file and 
specify a port on the command line.  This is helpful
if you don't have a configuration file yet need to 
specify an smtp server and a port.  The default port
is always port 25, so that if you don't specify this
option, and you don't have an email.conf file, it will
default to port 25 for you.

.TP
.B \-\-smtp-auth type | -m type
You must specify this option to use SMTP AUTH if you
aren't using a configuration file.  'type' is the type
of AUTH your server lets you use.  There are two supported
AUTH types with email: PLAIN and LOGIN.  

.TP
.B \-\-smtp-user user | -u user
If you don't want to keep your SMTP username in your
email.conf file (SMTP_AUTH_USER), then you can use this
command line option to specify it.

.TP
.B \-\-smtp-pass pass | -i pass
If you don't want to keep your SMTP password in your
email.conf file (SMTP_AUTH_PASS), then you can use this
command line option to specify it.

.TP
.B \-\-from-name name | -n name
This option will override MY_NAME in email.conf.
This is what the recipient will see in the from
field of their email client.

.TP
.B \-\-from-addr addr | -f addr
This option will override MY_EMAIL in email.conf.
This is what the recipient will see in the from
field of their email client as your email address.

.TP
.B \-\-conf-file file
Use this option to specify a configuration file other
than ~/.email.conf or /etc/email/email.conf.  

.TP
.B \-\-header string | \-H string,string,string,...
If you would like to append headers to the message 
that you are sending, you can use this option multiple
times and specify the header you would like to append
as well as delimit each header string with a comma for
multiple headers.

An example of a header string would look like:
--header "X-My-Header: Stuff goes here"

.TP
.B \-\-timeout secs | \-x secs
Set a timeout (in seconds) on the socket. This is
helpful if you would like to have the client stop
trying to send/recv data from the SMTP server if
it hangs after a period of time.

.TP
.B \-\-high-priority | \-o
This options allows you to take advantage of using
the priority option used by some email clients.
If the option is specified when sent to a user
using MS Outlook, a small exclimation mark will
be next to the message as to let the user know
this email is important.

.TP
.B \-\-html
This option allows you to send html emails. As
of right now, you need to write your own html.

.TP
.B \-\-gpg-pass , -g pass
If you don't want to keep your GPG password in your
email.conf file (GPG_PASS), then you can use this 
command line option to specify it.

.TP
.B \-\-tls
If you'd like to use TLS/SSL communication with the
smtp server, use this option (if it's not already
set in the config file.)

.TP
.B \-\-no-encoding
If you don't want eMail to automatically use UTF-8 encoding when finding
non ascii characters, use this option.

.SH CONFIGURATION
Configuration of email is fairly simple.  Just open
the default configuration file.  If you did not specify
an email home directory during your compilation of email
then this will be in /etc/email and the file
is called 'email.conf'.  The configuration options are
listed below.  


  SMTP_SERVER       : Server name, or IP
  SMTP_PORT         : Servers port number
  SENDMAIL_BIN      : Specify the sendmail binary path and options
  MY_NAME           : Specify your Name
  MY_EMAIL          : Specify your email address
  REPLY_TO          : Seperate reply to address
  SIGNATURE_FILE    : Your signature file
  SIGNATURE_DIVIDE  : A design for a divider
  ADDRESS_BOOK      : Location of your address book file
  SAVE_SENT_MAIL    : Directory to save email.sent file
  GPG_BIN           : Full path to gpg binary
  GPG_PASS          : Optional passphrase for gpg
  TEMP_DIR          : Specify where to store temp files
  SMTP_AUTH         : Specify what type of authentication 
                      for your smtp server.
  SMTP_AUTH_USER    : Specify a username login for SMTP AUTH
  SMTP_AUTH_PASS    : Specify a password for SMTP AUTH
  USE_TLS           : Boolean (true/false) if you want to use TLS/SSL
  VCARD             : Specify a vcard to attach to each message.
.br

You can choose to use sendmail instead of a remote smtp
server.  All you have to do is input the path of where
you have sendmail located in the SENDMAIL_BIN variable.
You may also provide options to pass sendmail.  The best
values for this would be /usr/lib/sendmail -t -i. If you
have both SENDMAIL_BIN and SMTP_SERVER uncommented and
both have values, then SMTP_SERVER will take presedence
over SENDMAIL_BIN and 'email' will try to contact the
SMTP server provided. 

You can create your own email.conf file in your home
directory.  It must be a hidden file.  For instance:
~/.email.conf

Email will look in your home directory before it looks
for a global configuration file.  The easiest thing
to do is simply copy your global config file to your
home directory as .email.conf and edit it as needed.

If you do not have a global config file or a personal
config file, email will choose defaults according to
your current login on the system. So a configuration
file is not manditory, it just helps you customize
email.

You can look at the basic configuration file for more
information on the configuration of 'email'.

.SH SMTP AUTH

Email now support SMTP AUTH.  I will briefly describe
how it works here.

To use email with an SMTP server that expects authentication
you must set a few options in your email.conf file. These
options are: SMTP_AUTH, SMTP_AUTH_USER, SMTP_AUTH_PASS.

SMTP_AUTH:
    This option must be set to one of the two: LOGIN or
    PLAIN. LOGIN and PLAIN are standard RFC compliant SMTP AUTH
    protocols.  If you are unsure which options to choose, 
    ask your ISP or SMTP Administrator if any of these are 
    supported. Usually it's a safe bet to use LOGIN for 
    SMTP AUTH.  Most AUTH servers support LOGIN.

SMTP_AUTH_USER:
    If you're using SMTP AUTH, please specify your username
    here.  This option is
.B MANDITORY
if you're using SMTP AUTH.

SMTP_AUTH_PASS:
    You can choose to set this option or not.  However, if
    you don't set this option, email will prompt you for your
    password before proceeding.  So if you're using email from
    a cron job and don't expect any interaction with email,
    please set this option!

USE_TLS:
    This option allows you to use TLS/SSL to communicate with
    your SMTP server. This is the most basic use of TLS/SSL
    possible. Please be advised that not much is done to 
    verify the certificate from the server. Also, currently
    there is no support for providing your own certificate for
    the server to authenticate since typical TLS does not do this.
    
.SH ENVIRONMENT VARIABLES

Two environment variables can be set that email will
check.  

EDITOR will allow you to specify your favorite
editor to use with email for constructing messages. If
this variable is not set, it will default to vi.  

TMPDIR can be set to specify a temporary directory to
place your temp files while email is working.  This is
analogous to the TEMP_DIR variable in email.conf.  
'email' will check TEMP_DIR first, then check your 
environment variable TMPDIR for a temporary directory.
If neither contains a value, email defaults to /tmp.


.SH ADDRESS BOOK

The address book for email takes on the format as
described below.  It will check for any syntax errors
in the address book and completely stop email if it
finds any, so try to keep the same gramatical syntax.
A single entry in the address book will look like:

  single: Tim = tim@somedomain.org   #Comments allowed

You are welcome to use a whole name with spaces as
long as you use a single quote closing the name
with another single quote.  You may also use double
quotes in place of the single quotes.  Example:

   single: 'Tim Gahan' = tim@somedomain.org

Groups are allowed and can only consist of comma
delimited single entries from the file and may
contain spaces.  You can *not* recursively specify
groups.  You may, however specify single email addresses
that are not part of the address book.

If you would like to break one line into two lines, you
should use the '\\' as a newline escape mark. Examples:

   group: Mygroup = 'Tim Gahan', John, Sam, Bob, \\
                       tookie@somedomain.org
                        
This example will specify a group with the single entries
of 'Dean Jones', John, Sam, Bob, and the unadded email
address of 'software@somedomain.org'.


.SH SIGNATURE FILE

The signature file is specified in the configuration
file.  You will also have the option to specify a
signature divider.  This is by default '---'.  This just
divides the signature file from the rest of the email.

There are a few wild cards that you can specify in
writing your signature file that will allow your signature
file to obtain dynamic options when sending your email.

.TP 
%v    
This will show the version of email
.TP 
%t    
This will show the time the email was sent
.TP 
%d    
This will display the date the email was sent
.TP 
%c    
This will display a completely formated date and time
.TP 
%h    
This will display the host type
.TP 
%f    
This will display the output of the 'fortune' command
.SH SIGNATURE EXAMPLE
Example of a signature file is below:

    This email was sent with 'email %v'
    Sent on host: %h
    At the time : %c
    Your fortune for today is:
    %f

This will end up replacing the %v, %h, %c, and %f with
their corresponding equivalents above respectively.


.SH EXAMPLES OF USAGE

.TP 
# Redirect your message to email
email \-s "Sending this.txt" tim@somedomain.org < this.txt
.TP 
# Make your message with email
email \-s "Subject" tim@somedomain.org
.TP 
# Encrypt and email to "Dean Jones"
email \-s "Encryted" \-encrypt tim@somedomain.org
.TP 
# Multiple Recipients from address book
email \-s "no subject" Dean,Jeff,Tom,Bob \-cc Josh,Carl
.TP
# Attach some files with your message
email \-s "files attached" Dean,Jeff --attach stuff.tar.gz,readme.doc

.SH BUGS

If you find any in this program, please submit them to 
.TP
.B http://www.cleancode.org/projects/email/contact

.SH AUTHORS

.B Dean Jones   - http://www.cleancode.org/projects/email/contact

.SH THANKS FOR RELEASE V3.0
.B Philip Lewis - Helped in debugging.

.SH COPYRIGHT

.B (C) 2001 - 2008



release/src/router/email-3.1.3/src/email.c
/**
    eMail is a command line SMTP client.

    Copyright (C) 2001 - 2008 email by Dean Jones
    Software supplied and written by http://www.cleancode.org

    This file is part of eMail.

    eMail is free software; you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation; either version 2 of the License, or
    (at your option) any later version.

    eMail is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.

    You should have received a copy of the GNU General Public License
    along with eMail; if not, write to the Free Software
    Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
**/
#if HAVE_CONFIG_H
# include "config.h"
#endif

#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <signal.h>

#include "getopt.h"

#include "email.h"
#include "conf.h"
#include "utils.h"
#include "addy_book.h"
#include "file_io.h"
#include "message.h"
#include "error.h"
#include "mimeutils.h"

static void
defaultDestr(void *ptr)
{
	xfree(ptr);
}

static struct option Gopts[] = {
	{"attach", 1, 0, 'a'},
	{"attachment", 1, 0, 'a'},
	{"verbose", 0, 0, 'v'},
	{"high-priority", 0, 0, 'o'},
	{"encrypt", 0, 0, 'e'},
	{"subject", 1, 0, 's'},
	{"sub", 1, 0, 's'},
	{"smtp-server", 1, 0, 'r'},
	{"smtp-port", 1, 0, 'p'},
	{"conf-file", 1, 0, 'c'},
	{"check-config", 0, 0, 't'},
	{"version", 0, 0, 'v'},
	{"blank-mail", 0, 0, 'b'},
	{"smtp-user", 1, 0, 'u'},
	{"smtp-username", 1, 0, 'u'},
	{"smtp-pass", 1, 0, 'i'},
	{"smtp-password", 1, 0, 'i'},
	{"smtp-auth", 1, 0, 'm'},
	{"gpg-pass", 1, 0, 'g'},
	{"gpg-password", 1, 0, 'g'},
	{"from-addr", 1, 0, 'f'},
	{"from-name", 1, 0, 'n'},
	{"header", 1, 0, 'H'},
	{"timeout", 1, 0, 'x'},
	{"html", 0, 0, 1},
	{"sign", 0, 0, 2},
	{"clearsign", 0, 0, 2},
	{"cc", 1, 0, 3},
	{"bcc", 1, 0, 4},
	{"to-name", 1, 0, 5},
	{"tls", 0, 0, 6},
	{"no-encoding", 0, 0, 7},
	{NULL, 0, NULL, 0 }
};

/**
 * Usage prints helpful usage information,
**/
void
usage(void)
{
	fprintf(stderr,
	    "Options information is as follows\n"
	    "email [options] recipient1,recipient2,...\n\n"
	    "    -h, -help module          Print this message or specify one of the "
	    "below options\n"
	    "    -V, -verbose              Display mailing progress.\n"
	    "    -f, -from-addr            Senders mail address\n"
	    "    -n, -from-name            Senders name\n"
	    "    -b, -blank-mail           Allows you to send a blank email\n"
	    "    -e, -encrypt              Encrypt the e-mail for first recipient before "
	    "sending\n"
	    "    -s, -subject subject      Subject of message\n"
	    "    -r, -smtp-server server   Specify a temporary SMTP server for sending\n"
	    "    -p, -smtp-port port       Specify the SMTP port to connect to\n"
	    "    -a, -attach file          Attach file and base64 encode\n"
	    "    -c, -conf-file file       Path to non-default configuration file\n"
	    "    -t, -check-config         Simply parse the email.conf file for errors\n"
	    "    -x, -timeout              Set socket timeout.\n"
	    "        -cc email,email,...   Copy recipients\n"
	    "        -bcc email,email,...  Blind Copy recipients\n"
	    "        -sign                 Sign the email with GPG\n"
	    "        -html                 Send message in HTML format "
	    "( Make your own HTML! )\n"
	    "        -tls                  Use TLS/SSL\n"
	    "    -m, -smtp-auth type       Set the SMTP AUTH type (plain or login)\n"
	    "    -u, -smtp-user username   Specify your username for SMTP AUTH\n"
	    "    -i, -smtp-pass password   Specify your password for SMTP AUTH\n"
	    "    -g, -gpg-pass             Specify your password for GPG\n"
	    "    -H, -header string        Add header (can be used multiple times)\n"
	    "        -high-priority        Send the email with high priority\n"
	    "        -no-encoding          Don't use UTF-8 encoding\n");

	exit(EXIT_SUCCESS);
}

char *
getConfValue(const char *tok)
{
	return (char *)dhGetItem(table, tok);
}

void
setConfValue(const char *tok, const char *val)
{
	dhInsert(table, tok, val);
}


/**
 * ModuleUsage will take an argument for the specified 
 * module and print out help information on the topic.  
 * information is stored in a written file in the location 
 * in Macro EMAIL_DIR. and also specified with EMAIL_HELP_FILE
 */
static void
moduleUsage(const char *module)
{
	FILE *help=NULL;
	short found=0;
	char *moduleptr=NULL;
	dstrbuf *buf = DSB_NEW;
	dstrbuf *help_file = expandPath(EMAIL_HELP_FILE);

	if (!(help = fopen(help_file->str, "r"))) {
		fatal("Could not open help file: %s", help_file->str);
		dsbDestroy(help_file);
		properExit(ERROR);
	}
	dsbDestroy(help_file);

	while (!feof(help)) {
		dsbReadline(buf, help);
		if ((buf->str[0] == '#') || (buf->str[0] == '\n')) {
			continue;
		}

		chomp(buf->str);
		moduleptr = strtok(buf->str, "|");
		if (strcasecmp(moduleptr, module) != 0) {
			while ((moduleptr = strtok(NULL, "|")) != NULL) {
				if (strcasecmp(moduleptr, module) == 0) {
					found = 1;
					break;
				}
			}
		} else {
			found = 1;
		}

		if (!found) {
			continue;
		}
		while (!feof(help)) {
			dsbReadline(buf, help);
			if (!strcmp(buf->str, "EOH\n")) {
				break;
			}
			printf("%s", buf->str);
		}
		break;
	}

	if (feof(help)) {
		printf("There is no help in the module: %s\n", module);
		usage();
	}

	dsbDestroy(buf);
	fclose(help);
	exit(0);
}


int
main(int argc, char **argv)
{
	int get;
	int opt_index = 0;          /* for getopt */
	char *cc_string = NULL;
	char *bcc_string = NULL;
	const char *opts = "f:n:a:p:oVedvtb?c:s:r:u:i:g:m:H:x:";

	/* Set certian global options to NULL */
	conf_file = NULL;
	global_msg = NULL;
	memset(&Mopts, 0, sizeof(struct mailer_options));
	Mopts.encoding = true;

	/* Check if they need help */
	if ((argc > 1) && (!strcmp(argv[1], "-h") ||
		!strcmp(argv[1], "-help") || !strcmp(argv[1], "--help"))) {
		if (argc == 3) {
			moduleUsage(argv[2]);
		} else if (argc == 2) {
			usage();
		} else {
			fprintf(stderr, "Only specify one option with %s: \n", argv[1]);
			usage();
		}
	}

	table = dhInit(28, defaultDestr);
	if (!table) {
		fprintf(stderr, "ERROR: Could not initialize Hash table.\n");
		exit(0);
	}

	while ((get = getopt_long_only(argc, argv, opts, Gopts, &opt_index)) > EOF) {
		switch (get) {
		case 'n':
			setConfValue("MY_NAME", xstrdup(optarg));
			break;
		case 'f':
			setConfValue("MY_EMAIL", xstrdup(optarg));
			break;
		case 'a':
			if (!Mopts.attach) {
				Mopts.attach = dlInit(defaultDestr);
			}
			dlInsertTop(Mopts.attach, xstrdup(optarg));
			break;
		case 'V':
			Mopts.verbose = true;
			break;
		case 'p':
			setConfValue("SMTP_PORT", xstrdup(optarg));
			break;
		case 'o':
			Mopts.priority = 1;
			break;
		case 'e':
			Mopts.gpg_opts |= GPG_ENC;
			break;
		case 's':
			Mopts.subject = optarg;
			break;
		case 'r':
			setConfValue("SMTP_SERVER", xstrdup(optarg));
			break;
		case 'c':
			conf_file = optarg;
			break;
		case 't':
			checkConfig();
			printf("Configuration file is proper.\n");
			dhDestroy(table);
			return (0);
			break;
		case 'v':
			printf("email - By: Dean Jones. Email: deanjones@cleancode.org.\n"
				"Version %s - Date %s \n", EMAIL_VERSION, COMPILE_DATE);
			dhDestroy(table);
			exit(EXIT_SUCCESS);
			break;
		case 'b':
			Mopts.blank = 1;
			break;
		case 'u':
			setConfValue("SMTP_AUTH_USER", xstrdup(optarg));
			break;
		case 'i':
			setConfValue("SMTP_AUTH_PASS", xstrdup(optarg));
			break;
		case 'm':
			setConfValue("SMTP_AUTH", xstrdup(optarg));
			break;
		case 'g':
			setConfValue("GPG_PASS", xstrdup(optarg));
			break;
		case 'H':
			if (!Mopts.headers) {
				Mopts.headers = dlInit(defaultDestr);
			}
			dlInsertTop(Mopts.headers, xstrdup(optarg));
			break;
		case 'x':
			setConfValue("TIMEOUT", xstrdup(optarg));
			break;
		case '?':
			usage();
			break;
		case 1:
			Mopts.html = 1;
			break;
		case 2:
			Mopts.gpg_opts |= GPG_SIG;
			break;
		case 3:
			cc_string = optarg;
			break;
		case 4:
			bcc_string = optarg;
			break;
		case 5:
			/* To name? */
			break;
		case 6:
			setConfValue("USE_TLS", xstrdup("true"));
			break;
		case 7:
			Mopts.encoding = false;
			break;
		default:
			/* Print an error message here  */
			usage();
			break;
		}
	}

	/* first let's check to make sure they specified some recipients */
	if (optind == argc) {
		usage();
	}

	configure();

	/* Check to see if we need to attach a vcard. */
	if (getConfValue("VCARD")) {
		dstrbuf *vcard = expandPath(getConfValue("VCARD"));
		if (!Mopts.attach) {
			Mopts.attach = dlInit(defaultDestr);
		}
		dlInsertTop(Mopts.attach, xstrdup(vcard->str));
		dsbDestroy(vcard);
	}

	/* set to addresses if argc is > 1 */
	if (!(Mopts.to = getNames(argv[optind]))) {
		fatal("You must specify at least one recipient!\n");
		properExit(ERROR);
	}

	/* Set CC and BCC addresses */
	if (cc_string) {
		Mopts.cc = getNames(cc_string);
	}
	if (bcc_string) {
		Mopts.bcc = getNames(bcc_string);
	}

	signal(SIGTERM, properExit);
	signal(SIGINT, properExit);
	signal(SIGPIPE, properExit);
	signal(SIGHUP, properExit);
	signal(SIGQUIT, properExit);

	createMail();
	properExit(0);

	/* We never get here, but gcc will whine if i don't return something */
	return 0;
}

